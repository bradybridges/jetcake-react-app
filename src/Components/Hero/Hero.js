import React from 'react';
import './Hero.scss';
const hero = require('../../Assets/hero.jpg');

export default function Hero({profile}) {
  return (
    <section id="hero-container">
      {profile && <h3 id="welcome-banner">Welcome, {profile.email}!</h3>}
      <h2 id="hero-header">This guy is my hero</h2>
        <img id="hero-img" src={hero} alt="Elon Musk" />
      <p id="description">Elon Musk always seems to be pushing the limits of technology, whether it be creating the most advanced electric vehicles, designing reusable rockets, or building underground highways. Elon Musk has had a hand in some of the most exciting innovations of modern times. He is one of the most notable engineers and entrepreneurs of recent times. A few of the qualities of Mr. Musk I respect the most are his drive to accomplish difficult problems, his creative and experimental attitude, and his ability to manage working on several projects at once. I make it a point to translate similar mentalities to my work as a developer. By breaking up challenging problems into smaller, more manageable ones, I can handle demanding tasks more effectively. And by following a strict schedule, I am easily able to work on and complete several projects simultaneously by their specified deadlines. And by continuously learning about the most modern front end technologies, I can create stunning applications that push the envelope of the frameworks that they harness.</p>
    </section>
  );
}
