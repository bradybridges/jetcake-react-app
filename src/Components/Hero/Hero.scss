#hero-container {
  height: 80vh;
  padding: 20px;
  overflow: scroll;
  width: 75%;
  margin: 0 auto;
}

#hero-container::-webkit-scrollbar {
  display: none;
}

#hero-img {
  width: 300px;
  border-radius: 4px;
}

#hero-header {
  font-size: 2em;
  color: white;
  background-color: rgba(0, 0, 0, 0.315);
  padding: 10px;
  margin: 10px 0px;
}

#description {
  color: white;
  background-color: rgba(0, 0, 0, 0.315);
  padding: 20px;
  width: 75%;
  margin: 0 auto;
  border-radius: 4px;
}

#welcome-banner {
  width: 100%;
  font-size: 2em;
  color: white;
  font-weight: bold;
  background-color: rgba(0, 0, 0, 0.315);
  margin: 10px auto;
  border-radius: 4px;
  padding: 10px;
}

@media only screen and (max-width: 900px) {
  #hero-container {
    width: 100%;
    padding: 0;
  }

  #description {
    width: 90%;
  }

}

@media only screen and (max-width: 450px) {
  #hero-img {
    width: 90%;
  }

  #description {
    width: 100%;
  }

  #hero-header {
    font-size: 2em;
  }

  #welcome-banner {
    font-size: 1.5em;
  }
}