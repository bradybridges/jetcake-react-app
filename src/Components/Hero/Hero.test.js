import React from 'react';
import { shallow } from 'enzyme';
import Hero from './Hero';

describe('Hero', () => {
  it('should match snapshot', () => {
    const mockProfile = {
      email: 'test@test.com',
      address: '123 st',
      dob: '1990-01-01',
      phone: '1112223333',
      securityOne: 'a',
      securityTwo: 'b',
      securityThree: 'c',
    };
    const wrapper = shallow(
      <Hero profile={mockProfile} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
