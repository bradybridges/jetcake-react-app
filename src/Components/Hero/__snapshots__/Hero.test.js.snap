// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero should match snapshot 1`] = `
<section
  id="hero-container"
>
  <h3
    id="welcome-banner"
  >
    Welcome, 
    test@test.com
    !
  </h3>
  <h2
    id="hero-header"
  >
    This guy is my hero
  </h2>
  <img
    alt="Elon Musk"
    id="hero-img"
    src="hero.jpg"
  />
  <p
    id="description"
  >
    Elon Musk always seems to be pushing the limits of technology, whether it be creating the most advanced electric vehicles, designing reusable rockets, or building underground highways. Elon Musk has had a hand in some of the most exciting innovations of modern times. He is one of the most notable engineers and entrepreneurs of recent times. A few of the qualities of Mr. Musk I respect the most are his drive to accomplish difficult problems, his creative and experimental attitude, and his ability to manage working on several projects at once. I make it a point to translate similar mentalities to my work as a developer. By breaking up challenging problems into smaller, more manageable ones, I can handle demanding tasks more effectively. And by following a strict schedule, I am easily able to work on and complete several projects simultaneously by their specified deadlines. And by continuously learning about the most modern front end technologies, I can create stunning applications that push the envelope of the frameworks that they harness.
  </p>
</section>
`;
